### PR TITLE
Aleksei/analytics

### LINKED ISSUE #3537: Analytics Pipeline
https://docs.google.com/document/d/1-MbVgKgcgoIEYL9auo0ECf43oo_FeYXDYh9Rd7amhAI/edit

### PR DESCRIPTION
**Pull request by iambeone**
Closes #3537 

Added custom metrics and some events data to store data

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: src/ActionModal/components/ActionModal.vue [hunk 1]
```diff
@@ -705,6 +707,17 @@ export default {
         this.selectedSignMethod
       )
       this.$emit(`txIncluded`)
+      // sending to ga
+      this.sendEvent(
+        {
+          network: this.network,
```
Comment (faboweb) [inline src/ActionModal/components/ActionModal.vue]: nice

File: src/components/common/TmBalance.vue [hunk 2]
```diff
@@ -248,6 +250,30 @@ export default {
       },
       /* istanbul ignore next */
       update(data) {
+        if (!this.sentToGA) {
```
Comment (faboweb) [inline src/components/common/TmBalance.vue]: why only once?

File: src/scripts/google-analytics.js [hunk 3]
```diff
@@ -1,46 +1,75 @@
...
+module.exports.deanonymize = function deanonymize() {
+  module.exports.anonymize(false)
+}
+
+module.exports.disableGoogleAnalytics = function disableGoogleAnalytics(gaUID) {
+  // if set to true disables google analytics
+  window[`ga-disable-${gaUID}`] = true
+}
+
+function customToNum(custom) {
```
Comment (faboweb) [inline src/scripts/google-analytics.js]: what does this do?

File: src/components/common/TmBalance.vue (hunk 2 above)
Comment (iambeone) [reply src/components/common/TmBalance.vue]: Otherwise, it will send it every 5 seconds.

File: src/scripts/google-analytics.js (hunk 3 above)
Comment (iambeone) [reply src/scripts/google-analytics.js]: GA takes parameters like dimension1 or dimension2. It's not so clear to understand in code, that's why I decided to replace them with more reliable words like `address` and `network`, so we can pass parameter like address = 'sdkjashkd' and not dimension1 = ' sdkjashkd'. This code replaces those 'friendly' parameters to google once.

Comment (faboweb) [reply src/scripts/google-analytics.js]: comment this!

Comment (faboweb) [reply src/scripts/google-analytics.js]:
```suggestion
/*
GA takes parameters like dimension1 or dimension2. This replaces the human readable values with the by GA required values.
*/
function customToNum(custom) {
```

File: src/components/common/TmBalance.vue (hunk 2 above)
Comment (faboweb) [reply src/components/common/TmBalance.vue]: ah right, not a problem though I guess ^^

**Comment by faboweb** [comment]
coverage is lacking

File: src/ActionModal/components/ActionModal.vue [hunk 4]
```diff
@@ -705,6 +707,17 @@ export default {
         this.selectedSignMethod
       )
       this.$emit(`txIncluded`)
+      // sending to ga
+      this.sendEvent(
```
Comment (faboweb) [inline src/ActionModal/components/ActionModal.vue]: what exactly does this event track?

Comment (iambeone) [reply src/ActionModal/components/ActionModal.vue]: This event tracks all actions with sending denoms like claiming rewards, sending, deligating, undeligating and redeligating. All actions, related to account amount change

Comment (jbibla) [reply src/ActionModal/components/ActionModal.vue]: should we remove the individual events in each action modal, then?

File: src/components/common/TmBalance.vue [hunk 5]
```diff
@@ -248,6 +250,30 @@ export default {
...
+          // sending to ga only once
+          sendEvent(
+            {
+              network: this.network,
+              address: this.address
+            },
+            "Portfolio",
+            "Balance",
+            "liquidStake",
+            data.overview.liquidStake
```
Comment (jbibla) [inline src/components/common/TmBalance.vue]: should we just include the whole overview object?

would be nice to have rewards balance as well - so we can know unclaimed rewards.

Comment (iambeone) [reply src/components/common/TmBalance.vue]: We can't include object itself, cause it heeds to be a number, but we can add rewards as well

File: src/ActionModal/components/ActionModal.vue (hunk 4 above)
Comment (iambeone) [reply src/ActionModal/components/ActionModal.vue]: Are we using them right now?

Comment (jbibla) [reply src/ActionModal/components/ActionModal.vue]: yes. we can leave it in for now. no problem.

File: src/components/common/TmBalance.vue (hunk 5 above)
Comment (jbibla) [reply src/components/common/TmBalance.vue]: yeah, please add rewards.

**Comment by faboweb** [comment]
Can we somehow unify `track` and `sendEvent`? It feels confusing to have both.

**Comment by jbibla** [comment]
ana! testing guru! 💪

**Pull request merged**